cli/remote_build.go

```diff
@@ -570,44 +566,40 @@ func postToApi(path string, formData url.Values) (*http.Response, error) {
 	return res, nil
 }
 
-func managementLogUrl(url, org, label string) string {
-	wsUrlRegexp := regexp.MustCompile(`^wss?://[^/]+/builds/([0-9a-f-]+)/progress$`)
-
-	if wsUrlRegexp.MatchString(url) {
-		submatches := wsUrlRegexp.FindStringSubmatch(url)
-		build := ""
-		if len(submatches) > 1 {
-			build = submatches[1]
-		}
-		if len(build) == 0 {
-			return ""
-		}
-
-		u, err := ybconfig.ManagementUrl(fmt.Sprintf("/%s/%s/builds/%s", org, label, build))
-		if err != nil {
-			log.Errorf("Unable to generate App Url: %v", err)
+// buildIDFromLogURL returns the build ID in a build log WebSocket URL.
+//
+// TODO(ch2570): This should come from the API.
+func buildIDFromLogURL(u *url.URL) (string, error) {
+	// Pattern is /builds/ID/progress
+	const prefix = "/builds/"
+	const suffix = "/progress"
+	if !strings.HasPrefix(u.Path, prefix) || !strings.HasSuffix(u.Path, suffix) {
+		return "", fmt.Errorf("build ID for %v: unrecognized path", u)
+	}
+	id := u.Path[len(prefix) : len(u.Path)-len(suffix)]
+	for i := 0; i < len(id); i++ {
+		if id[i] == '/' {
+			return "", fmt.Errorf("build ID for %v: unrecognized path", u)
```

Comment:
Any reason not to use `strings.ContainsRune` here?

Reply:
Good call. Done.